typst-math-symbols

jan-mate opened the pull request:
All Typst math symbols to Unicode

smeech reviewed (commented):
A package of over 900 symbols in simple trigger/replace pairs.
No `shell`, `script` or insecure code.
Passes the automated checks.

The trigger keys could be `trigger:` rather than `triggers:` as none of them have multiple phrases. This would also shorten the file with two lines for each expansion rather than three.

Sadly, the \<tab> expansions make the package unusable for me, however, as the triggers with an even number of characters all fail in my editors, leaving the first character of the trigger phrase before the injected character. These vary if I change the length of my tab-stops, but some always fail. They do work in my terminal emulator, but It's a shame to preclude them from environments where people are likely to want to use them.

Would you kindly change them to something more conventional (probably a prefix character such as a semicolon `;` or colon `:`, as many of them are standard words), please?

Sorry to create work for you!

jan-mate commented:
Thanks for the review!

I updated the package to use the `trigger` (`not triggers`) syntax.

I’m was unsure how to implement a prefix, because there's many conflicts where the shorter trigger fires first, such as `:Omega` vs `:Omega.inv`. I don't like using `word: true`, because I would also have to change `default.yml` to use a `word_seperators` without `.`, and still I wouldn't be able to easily type "Ω.".
Instead I updated the triggers to have `\` as suffix.

jan-mate commented:
I’d like to merge it as is, so it stays aligned with the Typst symbol naming convention.
